moveit_core/constraint_samplers/src/default_constraint_samplers.cpp

```diff
@@ -507,6 +509,15 @@ bool constraint_samplers::IKConstraintSampler::samplePose(Eigen::Vector3d& pos,
     quat = Eigen::Quaterniond(ikq.rotation());
   }
 
+  if (need_eef_to_ik_tip_transform_)
+  {
+    // After sampling the pose needs to be transformed to the ik chain tip
+    Eigen::Affine3d ikq(Eigen::Translation3d(pos) * quat.toRotationMatrix());
+    ikq = ikq * eef_to_ik_tip_transform_;
+    pos = ikq.translation();
+    quat = Eigen::Quaterniond(ikq.rotation());
+  }
```

Comment:
I think this solution does fix the issue you have described however I feel that applying a transform to a sampled pose is beyond the scope of what this function is meant to do.  In my opinion,  the transform should be applied by the function that calls `samplePose` in order to find a valid IK..